### PR TITLE
update tests, solve OutOfBounds issue, turn to es6

### PR DESCRIPTION
**Pull request by sandrabosk**
@mjarraya Please check the updated tests. 
Find here the proposed solution: https://gist.github.com/sandrabosk/1dc434be87b4efc03238e53ffae1376f

### CODE REVIEW

File: starter-code/test/test.js [hunk 1]
```diff
@@ -1,73 +1,72 @@
...
       assert.equal(sl instanceof SortedList, true);
+      assert.equal(sl.length, 0);
+      assert(Array.isArray(sl.items))
     });
   });
 
-  describe('#add(x)', function() {
-    var sl;
-    beforeEach(function(){
+  describe('#add(x)', () => {
```
Comment (mjarraya) [inline starter-code/test/test.js]: This unit should also test the spec that the list of items is sorted

File: starter-code/test/test.js [hunk 2]
```diff
@@ -1,73 +1,72 @@
...
+      }, Error)
     });
+
+   
   });
 
-  describe('#add(x) and get(i)', function() {
-    var sl;
-    beforeEach(function(){
+  describe('#add(x) and get(i)', () =>  {
```
Comment (mjarraya) [inline starter-code/test/test.js]: these specs could be integrated in the unit that describes `add`

File: starter-code/test/test.js [hunk 3]
```diff
@@ -1,73 +1,72 @@
...
     });
 
-    it('should return the element in that position', function() {
-      var foo = 10;
-      for(let i=0; i<200; i++) {
-        sl.add(foo*i);
-        assert.equal(sl.get(i), foo*i);
-      }
+    it('should return an OutOfBounds exception if there is no element in that position', () => {
+      assert.throws(() => {
```
Comment (mjarraya) [inline starter-code/test/test.js]: `assert.throws(() => { sl.get(4); }, Error, "OutOfBounds");`

It's not relevant to create a condition for equality of the return value of the method with the string `"undefined"` -which should never be reached-. `assert.throws()` only needs the function to be called with the right parameters, and we can pass in a third parameter to `.throws()` the error's name

File: starter-code/test/test.js [hunk 4]
```diff
@@ -77,97 +76,90 @@ describe('SortedList', function() {
...
     });
-  });
 
-  describe('#min()', function() {
-    var sl;
-    beforeEach(function() {
-      sl = new SortedList();
+    it('should return an EmptySortedList exception if there is no elements in the array', () => {
+       assert.throws(() => {
+        if (sl.max([]) === 'undefined') {
```
Comment (mjarraya) [inline starter-code/test/test.js]: `assert.throws(() => { sl.max([])) }, Error, "EmptySortedList");`

File: starter-code/test/test.js [hunk 5]
```diff
@@ -77,97 +76,90 @@ describe('SortedList', function() {
...
+    it('should return the min (lowest) value in the array', () => {
       sl.add(10);
       sl.add(20);
-
       assert.equal(sl.min(), 10);
     });
+
+    it('should return an EmptySortedList exception if there is no elements in the array', () => {
+      assert.throws(() => {
+        if (sl.min([]) === 'undefined') {
```
Comment (mjarraya) [inline starter-code/test/test.js]: see above

File: starter-code/test/test.js [hunk 6]
```diff
@@ -77,97 +76,90 @@ describe('SortedList', function() {
...
     })
 
-    it('should return the average of elements in the array', function() {
-      for(i=0; i<101; i++){
-        sl.add(i*2);
-      }
-      assert.equal(sl.average(), 100);
+    it('should return an EmptySortedList exception if there are no elements', () => {
+      assert.throws(() => {
+        if (sl.avg([]) === 'undefined') {
```
Comment (mjarraya) [inline starter-code/test/test.js]: see above

File: starter-code/test/test.js (hunk 2 above)
Comment (sandrabosk) [reply starter-code/test/test.js]: This test requires using `add()` and `get()` but in the moment we define `add()`, we still don't have `get()` defined. That's why I kept it after. Maybe I am not thinking in the right direction. Please let me know if I am wrong. Thank you.

**Comment by sandrabosk** [comment]
- updated tests that required a change
- removed global mocha installation (it's already using 6.x.x)

- **pending**: should additional test on _add ()_ be merged with _add()_?

File: starter-code/test/test.js (hunk 2 above)
Comment (mjarraya) [reply starter-code/test/test.js]: The students implement `add()` without the sorting, and the tests pass. Then they don't necessarily see what was wrong with their initial implementation (does it come from the `add()` ? from the `get()` ?) and have to backtrack to implement the `add` method correctly.

**Pull request merged**